### ISSUE TITLE
[Feature request] Show all projects in table of `infracost comment`

### ISSUE THREAD

**Issue by syphernl** [infracost/infracost#1538]
My current usage:
```bash
infracost breakdown --config-file infracost.yml --format json --out-file infracost.json
infracost comment gitlab --path infracost.json --repo project/repo --merge-request 1234 --gitlab-token TOKEN
```

Infracost only posts an `All projects`  project with the combined costs. 
It would be better if the individual projects would be visible here as well as a total.

### Current situation

----

:moneybag: Infracost estimate: **monthly cost will not change**
| Project | Previous | New | Diff |
|---------|----------|-----|------|
| All projects | $100 | $100 | $0 |

2 projects have no cost estimate changes

----

### Desired situation

----

:moneybag: Infracost estimate: **monthly cost will not change**
| Project | Previous | New | Diff |
|---------|----------|-----|------|
| Project 1 | $25 | $25 | $0 |
| Project 2 | $75 | $75 | $0 |
| |
| All projects | $100 | $100 | $0 |

2 projects have no cost estimate changes

----

### Proposed solution
(Added by @alikhajeh1 from Infracost)

Add `--show-options=all-projects` to `infracost comment` and `infracost output`; the idea being that in the future we could add other things to it (it's not clear what those are yet)

**Comment by aliscott** [comment]
Thanks @syphernl. I think the reason for this is that those 2 projects have no diff, so they are hidden from the table and represented by the `2 projects have no cost estimate changes` message. If the projects did have price diffs then they should show in the table.

I'm wondering if there's a better way of presenting this or if we want to make it a flag? There's some existing discussion for this here: https://github.com/infracost/infracost/discussions/1159.

Let me know what you think.

**Comment by syphernl** [comment]
#1159 looks similar, but in the example of the first post I do see a project with no changes but it still shows up in the table. Or is this the case because the other project has changes, and it then shows all individual projects?

I am currently expanding our implementation of Infracost in order to provide our teams with better indication in to the costs of the individual environments as well as the total costs. A flag to show _all projects_ - whether they have changes or not - would be helpful here.

**Comment by aliscott** [comment]
> I do see a project with no changes but it still shows up in the table

I think this is because we still show projects that had changes (resources renamed, etc) but resulted in a zero cost diff. Whereas projects that had no changes to resources don't show.

> I am currently expanding our implementation of Infracost in order to provide our teams with better indication in to the costs of the individual environments as well as the total costs.

Thanks for the context. This makes sense to me, there's benefit in seeing the cost per project even if there's no diff.

I think to support this we'd probably want to make it a flag, since there are cases where users don't want to see all project and only want to see what has changed. I'll add this to our weekly grooming meeting to see if the team has any ideas.

**Comment by alikhajeh1** [comment]
@syphernl I updated the issue description with the two solutions we can think of, which do you prefer? I'm not sure when we'll get to this yet but it seems like a good option to have (terragrunt users had previously told us to hide the projects with 0 diff from the table and detailed output as otherwise the list gets very long).

**Comment by syphernl** [comment]
>  which do you prefer? 

Option 2 is probably better for future expansion. 

One of which could be one to _optionally_ hide "free" projects (e.g. useful for Terragrunt where one of the projects may be a bunch of IAM roles which are free). Not to be confused with 0-diff projects.

**Comment by lvkins** [comment]
Agreed, current comment is not very informative in terms of costs breakdown per project if no changes were made. Given that we process so many details but output very little is unfortunate and should certainly be more configurable. Option 2 sounds reasonable, though we might also make use of the toggle block to display more detailed estimates breakdown. What you think?

**Comment by alikhajeh1** [comment]
@lvkins sorry I missed your comment - I updated the issue description with Option 2 as the proposed solution.

**Comment by alikhajeh1** [comment]
Hi @mohamadrhosseini - thanks for sharing the use-case. That makes sense.
I've added this to the project board and we'll try to get to it in September unless someone from the community wants to work on it, it shouldn't be that difficult but it will require golang experience.

**Comment by alikhajeh1** [comment]
@fatihtokus just wondering if there was any update on this?

**Comment by fatihtokus** [comment]
@alikhajeh1  created a PR.

**Comment by alikhajeh1** [comment]
Given the comments above (that the current behavior is confusing), I'm wondering if we should add the new `--show-options=all-projects` flag to all of our [CI/CD examples](https://www.infracost.io/docs/integrations/cicd/) so users are not confused about missing projects but are also aware that they can change that to make the summary table less noisy?

**Comment by aliscott** [comment]
I've updated this so we actually go with the `--show-all-projects` flag instead of the `--show-options` flag. After looking through the usage docs of the proposed changes `--show-all-projects` seems more consistent with our existing flags, since we already have `--show-skipped`. We can add other options as additional flags in the future if required.

This flag will go out in the v0.10.14 version.

**Comment by aliscott** [comment]
@syphernl, @lvkins, @rottenbytes, @visheyra, @mrhr7, @z0rc, @pacorreia this is now released in v0.10.14. You can use it by adding the `--show-all-projects` flag to any existing `infracost comment` commands in your CI pipelines. We'll add some more examples to our individual integration documents soon.

Thank you @fatihtokus for contributing this! 🙏
